File: internal/adapters/http/dto/cashflow.go
package dto

type CreateCashFlowRequest struct {
	Date       string  `json:"date"` // YYYY-MM-DD
	CategoryID int32   `json:"category_id"`
	Direction  string  `json:"direction"`
	Title      string  `json:"title"`
	Amount     float64 `json:"amount"`
	IsFixed    bool    `json:"is_fixed"`
}

type CashFlowResponse struct {
	ID         int32   `json:"id"`
	Date       string  `json:"date"`
	CategoryID int32   `json:"category_id"`
	Direction  string  `json:"direction"`
	Title      string  `json:"title"`
	Amount     float64 `json:"amount"`
	IsFixed    bool    `json:"is_fixed"`
}

type MonthlySummaryResponse struct {
	TotalIncome  float64 `json:"total_income"`
	TotalExpense float64 `json:"total_expense"`
	Balance      float64 `json:"balance"`
}

type CategorySummaryResponse struct {
	CategoryName string  `json:"category_name"`
	Direction    string  `json:"direction"`
	TotalAmount  float64 `json:"total_amount"`
}

type CopyFixedRequest struct {
	FromMonth string `json:"from_month"`
	ToMonth   string `json:"to_month"`
}

File: frontend/layers/cashflows/types/cashflow.ts
export type Direction = 'IN' | 'OUT'

export interface CashflowEntry {
  id: number
  date: string
  category_id: number
  category_name?: string
  payment_method_id: number
  payment_method_name?: string
  direction: Direction
  title: string
  amount: number
  is_fixed: boolean
  reversal_of_entry_id?: number | null
}

export interface CreateCashflowRequest {
  date: string
  category_id: number
  payment_method_id: number
  direction: Direction
  title: string
  amount: number
  is_fixed: boolean
}
